source/install/trouble_mysql.rst

```diff
@@ -0,0 +1,282 @@
+
+After creation of the database, attempt to restart the Mattermost server
+by navigating to the ``/opt/mattermost/bin`` directory and entering the
+command ``sudo -u mattermost ./platform``.
+
+**The mattermost Database Exists**
+
+If the ``mattermost`` database does exist, confirm that you have defined
+the database driver correctly in the
+``/opt/mattermost/config/config.json`` file. Open this file in a text
+editor, and review the value of ``"DataSource"``. It should be:
+
+ ::
+
+     "mmuser:*mmuser-password*@tcp(*host-name-or-IP*:3306)/mattermost?charset-utf8mb4,utf8&readTimeout-30s&writeTimeout-30s"
+
+You should also confirm that ``DriverName`` element (found immediately
+above the ``DataSource`` element) is set to ``mysql``.
+
+If you correct an error, restart the Mattermost server by navigating to
+the ``/opt/mattermost/bin`` directory and entering the command
+``sudo -u mattermost ./platform``.
+
+The Database User
+-----------------
+
+During installation you create a MySQL database user from the *mysql*
+prompt by issuing the command
+:samp: `create user 'mmuser'@'%' identified by '{mmuser-password}';`. The
+``mmuser-password`` value is a placeholder for the password you chose.
+You may also have specified an IP address rather than the wild card
+``%``.
+
+     **Note**
+
+    A MySQL user is fully defined by their username and the host that
+    they access MySQL from. These elements are separated by the ``@``
+    sign. The ``%`` character is a wild card indicating that the user
+    can access MySQL from any IP address. If the user you created
+    accesses MySQL from a specific IP address such as ``10.10.10.2``,
+    please adjust your actions accordingly.
+
+If the user and host combination that you created does not exist, you
+will see an error such as:
+
+ ::
+
+    [2017/09/20 17:06:18 EDT] [INFO] Pinging SQL master database
+    [2017/09/20 17:06:18 EDT] [EROR] Failed to ping DB retrying in 10 seconds 
+    err-Error 1045: Access denied for user 'mmuser'@'localhost' (using password: YES)
+
+ **Checking that mmuser Exists**
+
+ To check that this user exists log in to MySQL as root:
+``mysql -u root -p``.
+
+When prompted, enter the root password that you created when installing
+MySQL. From the ``mysql`` prompt enter the command
+``select User, Host from mysql.user;``. You should see something
+like the following
+
+ ::
+
+    +------------------+-----------+
+    | User             | Host      |
+    +------------------+-----------+
+    | mmuser           | %         |
+    | debian-sys-maint | localhost |
+    | mysql.session    | localhost |
+    | mysql.sys        | localhost |
+    | root             | localhost |
+    +------------------+-----------+
+    5 rows in set (0.00 sec)
+
+ **User Doesn't Exist**
+
+If ``'mmuser'@'%'`` does not exist, create this user by logging into
+MySQL as root and issuing the command: 
+:samp: `create user 'mmuser'@'%' identified by '{mmuser-password}';`.
```

Comment:
Not sure if this is meant to be a link or a code block.

Reply:
I think code block should be fine.